### PR TITLE
Get note

### PR DESCRIPTION
**Pull request by kristina71**

### CODE REVIEW

File: pkg/db/storage.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package db
```
Comment (RattusPetrucho) [inline pkg/db/storage.go:2]: Implement get note by id from db - Why from DB?

File: pkg/db/storage.go [hunk 2]
```diff
@@ -32,3 +34,24 @@ func (s *Storage) Get(ctx context.Context) ([]models.Note, error) {
 
 	return notes, errors.WithStack(err)
 }
+
+func (s *Storage) GetById(ctx context.Context, note models.Note) (models.Note, error) {
+	query, args, err := squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar).Select("id", "text", "created_time", "updated_time").From(tableName).Where(squirrel.Eq{"id": note.Id}).ToSql()
+	if err != nil {
+		log.Println(err)
```
Comment (RattusPetrucho) [inline pkg/db/storage.go:41]: Maybe better to log on top layer?

File: pkg/db/storage.go [hunk 3]
```diff
@@ -32,3 +34,24 @@ func (s *Storage) Get(ctx context.Context) ([]models.Note, error) {
...
+		return models.Note{}, err
```
Comment (RattusPetrucho) [inline pkg/db/storage.go:42]: Could we add stack trace to error?

File: pkg/db/storage.go [hunk 4]
```diff
@@ -32,3 +34,24 @@ func (s *Storage) Get(ctx context.Context) ([]models.Note, error) {
...
+	}
+
+	note_ := models.Note{}
+	err = s.db.Get(&note_, query, args...)
+
+	if err == sql.ErrNoRows {
```
Comment (RattusPetrucho) [inline pkg/db/storage.go:48]: Do we really what return common error for not found?

Comment (kristina71) [reply pkg/db/storage.go]: fixed

File: pkg/db/storage.go (hunk 3 above)
Comment (kristina71) [reply pkg/db/storage.go]: fixed

File: pkg/db/storage.go (hunk 2 above)
Comment (kristina71) [reply pkg/db/storage.go]: fixed

File: pkg/db/storage.go (hunk 1 above)
Comment (kristina71) [reply pkg/db/storage.go]: fixed

File: pkg/endpoints/endpoints.go [hunk 5]
```diff
@@ -48,3 +52,40 @@ func (e endpoint) GetNotes(w http.ResponseWriter, r *http.Request) {
 
 	tpl.Render(w, r, data, "notesHtml")
 }
+
+func (e endpoint) GetNote(w http.ResponseWriter, r *http.Request) {
+	url := models.Note{}
+	str := strings.Trim(r.URL.Path, "/")
```
Comment (RattusPetrucho) [inline pkg/endpoints/endpoints.go:58]: Can we use mux.Vars instead?

File: pkg/endpoints/endpoints.go [hunk 6]
```diff
@@ -48,3 +52,40 @@ func (e endpoint) GetNotes(w http.ResponseWriter, r *http.Request) {
...
+
+	re, _ := regexp.Compile(`\d+`)
+	res := re.FindAllString(str, -1)
+
+	value, err := strconv.Atoi(res[0])
+	if err != nil {
+		log.Fatal(err)
```
Comment (RattusPetrucho) [inline pkg/endpoints/endpoints.go:65]: Maybe we should notify user about error?

File: pkg/endpoints/endpoints.go [hunk 7]
```diff
@@ -48,3 +52,40 @@ func (e endpoint) GetNotes(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+	url.Id = uint16(value)
+
+	note, err := e.service.GetNote(r.Context(), url)
```
Comment (RattusPetrucho) [inline pkg/endpoints/endpoints.go:70]: Can we separate not found and internal errors?

File: pkg/endpoints/endpoints.go (hunk 5 above)
Comment (kristina71) [reply pkg/endpoints/endpoints.go:58]: No, we can't. We can't convert id to uint after that
For example,
```
vars := mux.Vars(r)
id := vars["id"]
url.Id = uint16(id)
```
it doesn't work in my way

File: pkg/endpoints/endpoints.go (hunk 6 above)
Comment (kristina71) [reply pkg/endpoints/endpoints.go:65]: You write to delete this in the previous comment - why do I have a mistake here than at all?

File: pkg/endpoints/endpoints.go (hunk 7 above)
Comment (kristina71) [reply pkg/endpoints/endpoints.go:70]: But we didn't implement different errors
It's not fair to demand it now)

File: pkg/endpoints/endpoints.go (hunk 5 above)
Comment (RattusPetrucho) [reply pkg/endpoints/endpoints.go:58]: is is string. We can always convert string to int with strconv package. 
https://pkg.go.dev/strconv#ParseUint

File: pkg/endpoints/endpoints.go (hunk 6 above)
Comment (RattusPetrucho) [reply pkg/endpoints/endpoints.go:65]: I didn't say delete user notifications...
We should tell user that there were an error and do not kill all server if we couldn't find note.

File: pkg/endpoints/endpoints.go (hunk 7 above)
Comment (RattusPetrucho) [reply pkg/endpoints/endpoints.go:70]: It's a wonderful tome for that!!!

**Review by RattusPetrucho — APPROVED** [review]

**Pull request merged**